### PR TITLE
add install requires to setup.py

### LINKED ISSUE #75: Add install_require and tests_require section to setup.py
**Describe the bug**
Setup.py currently lacks information for install_require and tests_require. Those sections should include the minimum information about which packages are needed to run the library as well as the packages to run the tests.

### PR DESCRIPTION
**Pull request by p-galligan**
Add minimum versions to the install requires. Fixes #75 

## Changes
Attempts to add minimum versions to the install_requires in the setup.py

## Questions
Am I missing anything here?
Do we even need anything for tests_requires? I believe that you only need to include packages that aren't part of the install requires or that aren't included in Python.

### CODE REVIEW

**Comment by p-galligan** [comment]
Additionally, re: our conversation about the difference between requirements and install_requires. The philosophical difference that I was seeing is that `install_requires` indicates what a project minimally needs to run. Meanwhile, the requirements file essentially lays out pip install arguments in a file.

https://packaging.python.org/discussions/install-requires-vs-requirements/

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,14 @@
...
       packages=find_packages(),
-      install_requires=[],
+      install_requires=["ArchivesSnake>=0.8.1",
+                        "boltons>=20.0.0",
+                        "fuzzywuzzy>=0.17.0",
+                        "pre-commit>=1.18.3"
+                        "pytest",
+                        "PyYAML>=5.2.0",
+                        "sphinx>=1.8.5",
+                        "tox>=3.14.0"],
```
Comment (helrond) [inline setup.py]:
```suggestion
      install_requires=["ArchivesSnake>=0.8.1",
                        "fuzzywuzzy>=0.17.0"],
```
IIRC, the install really only requires these two. The others are a) dependencies of ASnake or b) used in testing.

Comment (helrond) [reply setup.py]: It's also possible to not pin versions, and just let pip install the most recent version of the package. Something to think about.

File: setup.py [hunk 2]
```diff
@@ -13,7 +13,14 @@
...
       tests_require=[],
```
Comment (helrond) [inline setup.py]:
```suggestion
      tests_require=["pytest", "pre-commit", "tox", "sphinx"],
```

**Review by helrond — CHANGES_REQUESTED** [review]
Would suggest the following.

File: setup.py (hunk 2 above)
Comment (p-galligan) [reply setup.py]: Ohhh, ok. Those requirements make a little more sense to me now.

File: setup.py (hunk 1 above)
Comment (p-galligan) [reply setup.py]: @helrond Yeah, I saw that you can just include the package, but what I was reading is that it's best practice to include any known lower or upper bounds for your project, so I'd argue for keeping them in.

This section specifically: https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires

**Comment by bonniegee** [comment]
@p-galligan, can you create an issue and then link it to this PR?

File: setup.py (hunk 1 above)
Comment (helrond) [reply setup.py]: That sounds reasonable to me!

**Review by HaSistrunk — APPROVED** [review]

**Review by helrond — APPROVED** [review]

**Pull request merged**